### PR TITLE
Added Starship Prompt

### PR DESCRIPTION
**Pull request by open-sorcerer64**
Needs help in uninstall script

### CODE REVIEW

File: apps/Starship Prompt/install [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+mv *.ttf ~/.local/share/fonts/fira-code || error 'Failed to move fonts'
+mv *.otf ~/.local/share/fonts/fira-code || error 'Failed to move fonts'
+
+echo fontname >> ~/.config/lxterminal/lxterminal.conf
+sed -i '/fontname/c\fontname=Fira Code Medium Nerd Font 11' ~/.config/lxterminal/lxterminal.conf
+echo FontName >> ~/.config/xfce4/terminal/terminalrc
+sed -i '/FontName/c\FontName=Fira Code Medium Nerd Font 12' ~/.config/xfce4/terminal/terminalrc
+
+sh -c "$(curl -fsSL https://starship.rs/install.sh)" || error 'Failed to download starship prompt'
+echo  eval "$(starship init bash)" >> ~/.bashrc || error 'Failed to set starship prompt'
```
Comment (theofficialgman) [inline apps/Starship Prompt/install:18]: this does not work. did you test it? the eval attempts to run here, you want to place this at the end of the .bashrc, so you need to put it in ' '

```
echo 'eval "$(starship init bash)"' >> >> ~/.bashrc || error 'Failed to set starship prompt'
```
also this is dangerous if run twice, you should be removing all instances of this from .bashrc before adding it once

File: apps/Starship Prompt/uninstall [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+sh -c 'rm "$(command -v 'starship')"' || error 'Failed to remove starship'
+
+#HOW CAN I REMOVE TEXT FROM ~/.bashrc????
```
Comment (theofficialgman) [inline apps/Starship Prompt/uninstall:5]: in general, I would suggest not adding anything to .bashrc, but I see no way around it in this case

you can remove matching text like this
```
sed -i '/\eval "$(starship init bash)"/d' "$HOME/.bashrc"
```

File: apps/Starship Prompt/install (hunk 1 above)
Comment (theofficialgman) [reply apps/Starship Prompt/install:18]: see below post, you should do this when adding the line:

```
# remove all occurances
sed -i '/\eval "$(starship init bash)"/d' "$HOME/.bashrc"
# add one occurance to the .bashrc file
echo 'eval "$(starship init bash)"' >> ~/.bashrc || error 'Failed to set starship prompt'
```

**Comment by Itai-Nelken** [comment]
Using `.bashrc` isn't the best idea as not everyone uses bash.
I don't think it's a big problem yet, as people that use pi-apps most likely haven't changed their shell, but it might be a good idea to make an `api` function called `get_shell_rc_file_path` that gets the currently running shell's rc file.

**Comment by open-sorcerer64** [comment]
Yeah, it would be good idea as I too use `zsh` primarily

**Comment by theofficialgman** [comment]
> Using `.bashrc` isn't the best idea as not everyone uses bash. I don't think it's a big problem yet, as people that use pi-apps most likely haven't changed their shell, but it might be a good idea to make an `api` function called `get_shell_rc_file_path` that gets the currently running shell's rc file.

alternatively, just add it to all rc paths that are detected from a list. that way if zsh is installed at all, it will be added there

**Comment by open-sorcerer64** [comment]
Any ideas on how we can grab rc paths

**Comment by open-sorcerer64** [comment]
Uh ho

**Comment by theofficialgman** [comment]
@open-sorcerer64 I assume closing this was an accident?
I'll re-open

**Comment by open-sorcerer64** [comment]
> @open-sorcerer64 I assume closing this was an accident? I'll re-open

Yeah I actually messed up my fork ): thanks

**Comment by ryanfortner** [comment]
This isn't entirely necessary, but maybe we should consider using deb files for starship to make it easier to update for users.

I set up a github action on this repo: https://github.com/ryanfortner/starship-arm, but of course you can also download them from [my apt repo](https://apt.raspbian-addons.org/debian/pool/main/s/starship/)

**Comment by open-sorcerer64** [comment]
Continued in #1616

**Pull request closed without merging**